airbyte-integrations/connectors/source-retently/source_retently/source.py

```diff
@@ -66,12 +66,14 @@ def parse_response(
         stream_slice: Mapping[str, Any] = None,
         next_page_token: Mapping[str, Any] = None,
     ) -> Iterable[Mapping]:
-        data = response.json().get("data")
-        if data:
-            stream_data = data.get(self.json_path) if self.json_path else data
-            if stream_data:
-                for d in stream_data:
-                    yield d
+        data = response.json().get("data") or response.json()
+        stream_data = data.get(self.json_path) if self.json_path else data
+        if type(stream_data) == list :
```

Comment:
Same comment as above: since it seems that this is for only one specific stream (`nps`), I think it would be better to override the `parse_response` method of that specific stream.

Reply:
thanks @cptjacky
do you mean to do it in the parse_response method something like` if self.name == nps  yield x else yield y`?

Reply:
Hey @schlattk, I meant defining a different `parse_response` method on the NpsStream class itself (no `if` required in that situation)

Reply:
@cptjacky do you have an example of another source where it is done in this way?

Reply:
@schlattk https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-intercom/source_intercom/source.py#L342, the main class has a default `parse_response`, and then other streams define another `parse_response`

Reply:
ok @cptjacky I have implemented this now but some of the acceptance tests are failing not sure why as it doesn't seem to relate to changes I made. Would be interesting to see whether they also fail with the Airbyte test account credentials. Happy to run it if you can share the credentials

`Results (333.83s):
      23 passed
       6 failed
         - airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:109 TestSpec.test_match_expected[inputs0]
         - airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:121 TestSpec.test_enum_usage[inputs0]
         - airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:135 TestSpec.test_oneof_usage[inputs0]
         - airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:313 TestSpec.test_additional_properties_is_true[inputs0]
         - airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:330 TestConnection.test_check[inputs2]
         - airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:638 TestBasicRead.test_read[inputs0]`